### PR TITLE
feat(patient): add input validation

### LINKED ISSUE #1976: Create/Edit Patient inputs are not validated
When creating and editing a patient, only the required `givenName` is validated. The email and tel inputs can be anything and the date of birth can be a day greater than today.

## 🐛 Bug Report

Input fields from Editing/Creating Patients are not validated.

## To Reproduce

Input a text in the telephone field
Input a text that is not an email in the email field
Input a date in 2021 in the date of birth field

The components already give a way to add validation feedback to the user, but it is not used at the time.

### PR DESCRIPTION
**Pull request by LucasBN**
Fixes #1976

**Changes proposed in this pull request:**

- Prefix / Suffix / Family Name / Preferred Language Validated to ensure no numerical input
- Added to both edit and create
- Provides feedback to user
- Added tests for new code

### CODE REVIEW

**Comment by tehKapa** [comment]
@LucasBN can you resolve the 3 conflicts? thanks!

**Comment by LucasBN** [comment]
> @LucasBN can you resolve the 3 conflicts? thanks!

@tehKapa Done

File: src/__tests__/patients/patient-slice.test.ts [hunk 1]
```diff
@@ -203,6 +203,11 @@ describe('patients slice', () => {
       const expectedPatient = {
         id: expectedPatientId,
         givenName: undefined,
+        dateOfBirth: addDays(new Date(), 4).toISOString(),
+        suffix: '061002',
+        prefix: '061002',
+        familyName: '061002',
+        preferredLanguage: '061002',
```
Comment (jackcmeyer) [inline src/__tests__/patients/patient-slice.test.ts:210]: This test was changed to only check for required fields. I think that we should remove everything but `givenName` in this test. 

I think we'll need to add some other test which checks for numbers in the string.

Comment (jackcmeyer) [reply src/__tests__/patients/patient-slice.test.ts:210]: I believe this is why the build is failing.

File: src/__tests__/patients/patient-slice.test.ts [hunk 2]
```diff
@@ -386,6 +395,10 @@ describe('patients slice', () => {
         id: expectedPatientId,
         givenName: undefined,
         dateOfBirth: addDays(new Date(), 4).toISOString(),
+        suffix: '061002',
```
Comment (jackcmeyer) [inline src/__tests__/patients/patient-slice.test.ts:398]: let's change this test to only test the numbers in the values.

**Pull request merged**